### ISSUE TITLE
10_nvidia_wayland.json and 15_nvidia_gbm.json missing on Tumbleweed

### ISSUE THREAD

**Issue by sndirsch** [openSUSE/nvidia-driver-G06#26]
This looks wrong.

```
%if 0%{?suse_version} < 1550
# EGL driver config
mkdir -p %{buildroot}%{_datadir}/egl/egl_external_platform.d
install -m 644 10_nvidia_wayland.json 15_nvidia_gbm.json 20_nvidia_xcb.json 20_nvidia_xlib.json \
    %{buildroot}%{_datadir}/egl/egl_external_platform.d
%endif
```

First, it should be `0%{?sle_version} < 150X00` instead. But I figured out that 

`10_nvidia_wayland.json` and `15_nvidia_gbm.json` are missing from `libnvidia-egl-wayland` and `libnvidia-egl-gbm` sources. So as long as these are not added to the sources and installed during packaging, we need to change that.

```
# EGL driver config
mkdir -p %{buildroot}%{_datadir}/egl/egl_external_platform.d
install -m 644 10_nvidia_wayland.json 15_nvidia_gbm.json %{buildroot}%{_datadir}/egl/egl_external_platform.d
%if 0%{?sle_version} < 150700
install -m 644 20_nvidia_xcb.json 20_nvidia_xlib.json %{buildroot}%{_datadir}/egl/egl_external_platform.d
%endif
```

Same here:

```
%if 0%{?suse_version} < 1550
%dir %{_datadir}/egl
%dir %{_datadir}/egl/egl_external_platform.d
%{_datadir}/egl/egl_external_platform.d/10_nvidia_wayland.json
%{_datadir}/egl/egl_external_platform.d/15_nvidia_gbm.json
%{_datadir}/egl/egl_external_platform.d/20_nvidia_xcb.json
%{_datadir}/egl/egl_external_platform.d/20_nvidia_xlib.json
%endif
```

Should be instead:

```
%dir %{_datadir}/egl
%dir %{_datadir}/egl/egl_external_platform.d
%{_datadir}/egl/egl_external_platform.d/10_nvidia_wayland.json
%{_datadir}/egl/egl_external_platform.d/15_nvidia_gbm.json
%if 0%{?sle_version} < 150700
%{_datadir}/egl/egl_external_platform.d/20_nvidia_xcb.json
%{_datadir}/egl/egl_external_platform.d/20_nvidia_xlib.json
%endif
```

**Comment by scaronni** [comment]
Hello, the json files are inside the sources of the respective projects, just not in a release. The Nvidia driver always ships the latest tip of `egl-gbm`, `egl-wayland` and `egl-x11`. Unfortunately this is the way it's packaged, because inside of Nvidia those sources are integrated in the driver build.

In particular., the latest `egl-wayland` snapshot (post 1.1.16 contains lots of fixes for Wayland in general and in particular for KDE using a Wayland session.

These snapshots are what we ship for distributions with Wayland Explicit sync enabled, ex. Fedora:
- https://src.fedoraproject.org/rpms/egl-gbm/blob/rawhide/f/egl-gbm.spec
- https://src.fedoraproject.org/rpms/egl-wayland/blob/rawhide/f/egl-wayland.spec
- https://src.fedoraproject.org/rpms/egl-x11/blob/rawhide/f/egl-x11.spec

**Comment by scaronni** [comment]
I think Tumbleweed falls in the category of Explicit Sync enabled distributions ([Explicit Sync wayland procotol](https://github.com/NVIDIA/egl-wayland/blob/master/src/meson.build#L23) + Xwayland 24.1+).

**Comment by sndirsch** [comment]
This results in this change.

```
%if 0%{?sle_version} < 150700
# EGL driver config
mkdir -p %{buildroot}%{_datadir}/egl/egl_external_platform.d
install -m 644 10_nvidia_wayland.json 20_nvidia_xcb.json 20_nvidia_xlib.json \
  %{buildroot}%{_datadir}/egl/egl_external_platform.d
%if 0%{?sle_version} < 150500
install -m 644 15_nvidia_gbm.json \
  %{buildroot}%{_datadir}/egl/egl_external_platform.d
%endif
%endif
[...]
%if 0%{?sle_version} < 150700
%dir %{_datadir}/egl
%dir %{_datadir}/egl/egl_external_platform.d
%{_datadir}/egl/egl_external_platform.d/10_nvidia_wayland.json
%{_datadir}/egl/egl_external_platform.d/20_nvidia_xcb.json
%{_datadir}/egl/egl_external_platform.d/20_nvidia_xlib.json
%if 0%{?sle_version} < 150500
%{_datadir}/egl/egl_external_platform.d/15_nvidia_gbm.json
%endif
```

**Comment by sndirsch** [comment]
And this is now already considered done with PR #27 and my commit fix  f10d35e10da457b96c96b30fec202b8bbf2a6bc4  right afterwards.

**Comment by sndirsch** [comment]
Closing as done.
